test/libp2p/multi/cid_test.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <libp2p/common/hexutil.hpp>
+#include <libp2p/common/literals.hpp>
+#include <libp2p/multi/content_identifier.hpp>
+#include <libp2p/multi/content_identifier_codec.hpp>
+#include <libp2p/multi/multibase_codec/multibase_codec_impl.hpp>
+#include <libp2p/multi/multicodec_type.hpp>
+#include <libp2p/multi/uvarint.hpp>
+#include <testutil/outcome.hpp>
+
+using libp2p::multi::ContentIdentifier;
+using libp2p::multi::ContentIdentifierCodec;
+using libp2p::multi::HashType;
+using libp2p::multi::MultibaseCodec;
+using libp2p::multi::MultibaseCodecImpl;
+using libp2p::multi::MulticodecType;
+using libp2p::multi::Multihash;
+using libp2p::multi::UVarint;
+using libp2p::common::operator""_multihash;
+using libp2p::common::operator""_unhex;
+
+const Multihash ZERO_MULTIHASH =
+    "12200000000000000000000000000000000000000000000000000000000000000000"_multihash;
+const Multihash EXAMPLE_MULTIHASH =
+    "12206e6ff7950a36187a801613426e858dce686cd7d7e3c0fc42ee0330072d245c95"_multihash;
+
+TEST(CidTest, PrettyString) {
+  ContentIdentifier c1(ContentIdentifier::Version::V1, MulticodecType::IDENTITY,
+                       ZERO_MULTIHASH);
+  ASSERT_EQ(c1.toPrettyString("base58"),
+            "base58 - cidv1 - raw - sha2-256-256-"
+                + libp2p::common::hex_lower(ZERO_MULTIHASH.getHash()));
+  ContentIdentifier c2(ContentIdentifier::Version::V0, MulticodecType::DAG_PB,
+                       EXAMPLE_MULTIHASH);
+  ASSERT_EQ(c2.toPrettyString("base64"),
+            "base64 - cidv0 - dag-pb - sha2-256-256-"
+                + libp2p::common::hex_lower(EXAMPLE_MULTIHASH.getHash()));
```

Comment:
We can also add tests for CID's from spec: https://github.com/multiformats/cid#human-readable-cids

Reply:
We don't have a codec for base58btc and adding this is, as for me, is beyond this task.